### PR TITLE
FXIOS-1263 ⁃ Fixes #7602 - Search information count

### LINKED ISSUE #7602: FXIOS-1163 ⁃ Search information counts
**Vision**

As a PM, I want visibility into how often different search activities are taking place within our app to monitor for issues and measure results of improvements to the search workflows

**Acceptance Criteria**

* Counts in Glean for follow-on and organic search in Google
* Data is inline with Fenix for simpler reporting

**Notes**

* Corresponding [Fenix tickets and PRs](https://github.com/mozilla-mobile/fenix/issues/6448)
* [Search definitions](https://docs.telemetry.mozilla.org/datasets/search.html)
* Travis or Megan can help answer questions and ensure tracking is inline with Fenix for simpler reporting

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1163)

### PR DESCRIPTION
**Pull request by nbhasin2**
┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1263)

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift
Comment (nbhasin2) [inline Client/Frontend/Browser/BrowserViewController.swift:1835]: - when you tap on a suggestion

File: Client/Frontend/Browser/BrowserViewController.swift
Comment (nbhasin2) [inline Client/Frontend/Browser/BrowserViewController.swift:1602]: - when user enters a search word in url bar

**Review by dnarcese — APPROVED** [review]

**Review by nbhasin2 — COMMENTED** [review]
@dnarcese +1 
But since I can't approve it myself you go ahead and approve this PR. Feel free to merge it when you approve :)

**Comment by dnarcese** [comment]
# Request for data collection review form
1. What questions will you answer with this data?
These probes will be used to understand what percentage of our searches come from organic searches versus standard browser access points.
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
To determine what volume of searches are not monetising via Firefox.
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
No alternatives.
4. Can current instrumentation answer these questions?
Not currently.
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.
	
| Measurement Description	        | Data Collection Category           | Tracking Bug #  |
| ------------- |:-------------:| -----:|
| sap search performed      | 2 | https://github.com/mozilla-mobile/firefox-ios/issues/7602 |
| sap follow on search  performed  | 2      |   https://github.com/mozilla-mobile/firefox-ios/issues/7602 |
| organic search performed | 2      |    https://github.com/mozilla-mobile/firefox-ios/issues/7602 |
6. How long will this data be collected? Choose one of the following:
 I want this data to be collected for 6 months initially (potentially renewable).
7. What populations will you measure?
All users across all channels and locales.
8. If this data collection is default on, what is the opt-out mechanism for users?
Setting in Firefox iOS to opt out.
9. Please provide a general description of how you will analyze this data.
We will analyse the volume differences between organic and SAPs through Glean / Amplitude
10. Where do you intend to share the results of your analysis?
With the mobile team and data science teams.
11. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:
No

**Comment by dnarcese** [comment]
r+
# Data Review Form

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, telemetry is documented in `docs/metrics.md`

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, firefox iOS provides data controls in the app settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

This should be re-reviewed w/ @athomasmoz every 6mo

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Type 2

5) Is the data collection request for default-on or default-off?

Default on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does there need to be a check-in in the future to determine whether to renew the data? (Yes/No) (If yes, set a todo reminder or file a bug if appropriate)**

Should review all probes for necessity every 6mo w/ @athomasmoz 

9) Does the data collection use a third-party collection tool? 

No

File: Client/Frontend/Browser/Tab.swift
Comment (st3fan) [inline Client/Frontend/Browser/Tab.swift:608]: I'm reading this code because I'm moving it to Focus. This comparison here worries me because this does not just match `https://www.google.com`, it will also match `https://www.example.com/google-rocks` - basically any URL with `google` in it at any place. That will not lead to correct measurements.

Comment (nbhasin2) [reply Client/Frontend/Browser/Tab.swift:608]: Good catch, @st3fan I don't remember why we did it. This is only to set the provider and I feel it slipped as we don't support many providers and only google at the moment. This looks like something we should improve upon. Although we do have safety check currently implemented that should prevent any in-correct counts but please let me know if you encounter any scenario that have been missed or something I can improve upon.

A better solution: 
`url.domainURL.absoluteString` and check if the provider is google or we can do `url.baseDomain` with base domain google.

---
Safety Check (this is already in place) and the following method should stop unwanted google url captures.

https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Telemetry/SearchTelemetry.swift#L71

https://github.com/mozilla-mobile/firefox-ios/blob/0e5dca7f78be4d51c1cd6a68dc86fc9ef313ca14/Client/Telemetry/SearchTelemetry.swift#L85-L90

Comment (st3fan) [reply Client/Frontend/Browser/Tab.swift:608]: Yeah on `main` this looks good - I don't think search telemetry can trigger there because of the match that I mentioned above.

**Pull request merged**
